crates/expr/src/expr.rs

```diff
@@ -26,11 +26,32 @@ pub enum ProjectName {
 }
 
 impl ProjectName {
-    /// What is the [TableId] for this projection?
-    pub fn table_id(&self) -> Option<TableId> {
+    /// The [TableSchema] of the returned rows.
+    /// Note this expression returns rows from a relvar.
+    /// Hence it this method should never return [None].
+    pub fn return_table(&self) -> Option<&TableSchema> {
         match self {
-            Self::None(input) => input.table_id(None),
-            Self::Some(input, var) => input.table_id(Some(var.as_ref())),
+            Self::None(input) => input.return_table(),
+            Self::Some(input, alias) => input.find_table_schema(alias),
+        }
+    }
+
+    /// The [TableId] of the returned rows.
+    /// Note this expression returns rows from a relvar.
+    /// Hence it this method should never return [None].
+    pub fn return_table_id(&self) -> Option<TableId> {
+        match self {
+            Self::None(input) => input.return_table_id(),
+            Self::Some(input, alias) => input.find_table_id(alias),
+        }
+    }
+
+    /// Iterate over the returned column names and types
+    pub fn iter_return_fields(&self, mut f: impl FnMut(&str, &AlgebraicType)) {
```

Comment:
NIT: It hits me that `iter` is not the best name for these methods that implement internal iteration, it could be `each` instead, but this could be done later.